lib/shopify-cli/commands/help.rb

```diff
@@ -21,27 +21,45 @@ def call(args, _name)
         end
 
         preamble = <<~MESSAGE
-          CLI to help build Shopify apps faster.
-
           Use {{command:#{ShopifyCli::TOOL_NAME} help <command>}} to display detailed information about a specific command.
 
-          {{bold:Available commands}}
+          {{bold:Available core commands:}}
+
         MESSAGE
         @ctx.puts(preamble)
 
-        visible_commands = ShopifyCli::Commands::Registry
-          .resolved_commands
-          .select { |_name, c| !c.hidden }
-          .sort
+        core_commands.each do |name, klass|
+          next if name == 'help'
+          @ctx.puts("{{command:#{name}}}: #{klass.help}\n")
+        end
+
+        return unless Project.current_project_type
 
-        visible_commands.each do |name, klass|
+        @ctx.puts("{{bold:Available commands for #{project_name} projects:}}\n\n")
+
+        local_commands.each do |name, klass|
           next if name == 'help'
           @ctx.puts("{{command:#{name}}}: #{klass.help}\n")
         end
       end
 
       private
 
+      def project_name
+        ProjectType.load_type(Project.current_project_type).project_name
+      end
+
+      def core_commands
+        resolved_commands
+          .select { |_name, c| !c.hidden }
+          .select { |_name, c| c.to_s.include?('ShopifyCli::Commands') }
```

Comment:
I'm still thinking about a better way to distinguish core vs local commands.

Reply:
I'll be committing a `ShopifyCli::Commands.core_command?` method shortly as part of another fix, I'll add you as a reviewer

Reply:
Also, I'm thinking this should probably be extracted out of the `Command` class and into something like a registry, which could have query methods for the core and local commands. But that could be done in a follow-up PR.